packages/react-router/modules/Router.js

```diff
@@ -16,17 +16,30 @@ class Router extends React.Component {
 
   getChildContext() {
     return {
-      router: this.props.history
+      router: this.router
     }
   }
 
   componentWillMount() {
-    const { children } = this.props
+    const { children, history } = this.props
 
     invariant(
       children == null || React.Children.count(children) === 1,
       'A <Router> may have only one child element'
     )
+
+    this.router = {
+      ...history,
+      match: {
+        path: '/',
+        url: '/',
+        params: {}
+      }
```

Comment:
Yes. I initially wrote this where I would update `isExact` on every update. I could not think of a reason why it would be needed, though, so I ended up removing it. I'm sure someone will have their reasons, so it makes sense to include it.